pkg/gator/reader/filereader.go

```diff
@@ -143,15 +143,11 @@ func verifyFile(filename string) error {
 }
 
 func readStdin() ([]*unstructured.Unstructured, error) {
-	stdinfo, err := os.Stdin.Stat()
+	_, err := os.Stdin.Stat()
```

Comment:
it's okay to send an empty stream to `ReadK8sResources` since it can handle it

Reply:
What happens if you run the `gator` command by hand, without piping anything. Does it block until you terminate standard input (CTRL-D)?

Reply:
Great call out! In this commit sha, `gator test` did hang on stdin. I checked w the docs and it seems that writing in manifests specifically (aka not piping or redirecting) is not a supported use case.

I've added in a different way to check that data is being redirected or piped to stdin and read and a test to make sure that `gator test` doesn't hang.

Reply:
> it seems that writing in manifests specifically (aka not piping or redirecting) is not a supported use case.

three things should be explicitly supported:

`cat <file> | gator`

`gator -f <file>`

`cat <file> gator -f <path>`

Standard shell conventions should handle edge cases. E.g.:

`cat | gator` should allow copy/paste + CRL-D

[here docs](https://tldp.org/LDP/abs/html/here-docs.html) and input redirection (`<`) should also come for free with handling stdin.

One convention I'm not sure we get is `gator -f -`, where the dash indicates "read from stdin"